notebooks/304-pytorch-post-training-quantization/304-pytorch-post-training-quantization.ipynb

```diff
@@ -0,0 +1,698 @@
+   "source": [
+    "def create_dataloaders(batch_size=128):\n",
+    "    train_dir = DATASET_DIR / 'train'\n",
+    "    val_dir = DATASET_DIR / 'val' / 'images'\n",
+    "    normalize = transforms.Normalize(mean=[0.485, 0.456, 0.406],\n",
+    "                                     std=[0.229, 0.224, 0.225])\n",
+    "    train_dataset = datasets.ImageFolder(\n",
+    "        train_dir,\n",
+    "        transforms.Compose([\n",
+    "            transforms.Resize(IMAGE_SIZE),\n",
+    "            transforms.ToTensor(),\n",
+    "            normalize,\n",
+    "        ]))\n",
+    "    val_dataset = datasets.ImageFolder(\n",
+    "        val_dir,\n",
+    "        transforms.Compose([\n",
+    "            transforms.Resize(IMAGE_SIZE),\n",
+    "            transforms.ToTensor(),\n",
+    "            normalize\n",
+    "        ]))\n",
+    "\n",
+    "    train_loader = torch.utils.data.DataLoader(train_dataset, batch_size=batch_size, shuffle=True,\n",
+    "                                               num_workers=4, pin_memory=True, sampler=None)\n",
+    "    \n",
+    "    val_loader = torch.utils.data.DataLoader(val_dataset, batch_size=batch_size, shuffle=False,\n",
+    "                                             num_workers=4, pin_memory=True)\n",
+    "    return train_loader, val_loader\n",
+    "\n",
+    "\n",
+    "train_loader, val_loader = create_dataloaders()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "880f550a",
+   "metadata": {},
+   "source": [
+    "### I. Evaluate the loaded model"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "19c6d048",
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "acc1 = validate(val_loader, model)\n",
+    "print(f'Test accuracy of FP32 model: {acc1:.3f}')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "21a797fe",
+   "metadata": {},
+   "source": [
+    "### Export the FP32 model to ONNX, which is supported by OpenVINO™ Toolkit, to benchmark it in comparison with the INT8 model."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "db502e73",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "dummy_input = torch.randn(1, 3, *IMAGE_SIZE).to(device)\n",
+    "torch.onnx.export(model, dummy_input, fp32_onnx_path, opset_version=10, training=True)\n",
+    "print(f\"FP32 ONNX model was exported to {fp32_onnx_path}.\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "49d2f412",
+   "metadata": {},
+   "source": [
+    "## II. Create and initialize quantization\n",
+    "NNCF enables post-training quantization using trainig dataset for the initialization process of recently added quantization layers. The framework is designed so that modifications to your original training code are minor. Quantization is the simplest scenario and requires only 3 modifications.\n",
+    "\n",
+    "### 1. Configure NNCF parameters to specify compression"
```

Comment:
Same, make this paragraph plain-text instead of header-text. Use `1)` format instead of `1.` or, perhaps, don't use indexing here at all.

Reply:
Done